doc/before-after-reboot-checks.md

```diff
@@ -0,0 +1,48 @@
+# Before and After Reboot Checks
+
+CLUO provides optional functionality for running arbitrary actions before and
+after the rebooting of a node using node labels and annotations.
+
+## Configuring the `update-operator`
+
+When depolying the `update-operator`, you can provide a comma-separated list of
+node annotations to look for before a reboot should occur, and another one for
+after a reboot has occured. The list is provided as a command line argument to
+the binary -
+
+```bash
+update-operator --before-reboot-annotations anno1,anno2 --after-reboot-annotations anno3,anno4
+```
+
+## Making a check
+
+A check that needs to run before reboot should look for the
```

Comment:
Seems like we could be more concise:

Pods that should run before a reboot should node select on the label blah and set the required before-reboot annotation to allow the reboot to occur. Pods that should run after a reboot should node select on the label blah and annotate the node with the required after-reboot annotation to allow the node to become schedulable again. A DaemonSet with a [node selector](link) is the recommended way to achieve this.

* before-reboot-daemonset.yaml
* after-reboot-daemonset.yaml